include/RAJA/scan.hxx

```diff
@@ -0,0 +1,171 @@
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/config.hxx"
+
+#include <functional>
+#include <type_traits>
+
+namespace RAJA
+{
+
+/*!
+******************************************************************************
+*
+* \brief  inclusive in-place scan execution policy
+*
+******************************************************************************
+*/
+
+template <typename ExecPolicy,
+          typename Iter,
+          typename T,
+          typename BinaryFunction>
+void inclusive_scan_inplace(Iter begin, Iter end, BinaryFunction f, T v)
+{
+  inclusive_scan_inplace(ExecPolicy{}, begin, end, f, v);
+}
+
+template <typename ExecPolicy,
+          typename Iter,
+          typename T = typename std::iterator_traits<Iter>::value_type>
+void inclusive_scan_inplace(Iter begin, Iter end)
+{
+  inclusive_scan_inplace(ExecPolicy{}, begin, end, std::plus<T>{}, T{0});
+}
+
+/*!
+******************************************************************************
+*
+* \brief  inclusive in-place scan execution policy
+*
+******************************************************************************
+*/
+
+
+template <typename ExecPolicy,
+          typename Iter,
+          typename T,
+          typename BinaryFunction>
+void exclusive_scan_inplace(Iter begin, Iter end, BinaryFunction f, T v)
+{
+  exclusive_scan_inplace(ExecPolicy{}, begin, end, f, v);
+}
+
+template <typename ExecPolicy,
+          typename Iter,
+          typename T = typename std::iterator_traits<Iter>::value_type>
+void exclusive_scan_inplace(Iter begin, Iter end)
+{
+  exclusive_scan_inplace(ExecPolicy{}, begin, end, std::plus<T>{}, T{0});
+}
+
+/*!
+******************************************************************************
+*
+* \brief  inclusive scan execution policy
+*
+******************************************************************************
+*/
+
+template <typename ExecPolicy,
+          typename Iter,
+          typename IterOut,
+          typename BinaryFunction,
+          typename T>
+void inclusive_scan(Iter begin, Iter end, IterOut out, BinaryFunction f, T v)
+{
+  inclusive_scan(ExecPolicy{}, begin, end, out, f, v);
+}
+
+template <typename ExecPolicy,
+          typename Iter,
+          typename IterOut,
+          typename T = typename std::iterator_traits<Iter>::value_type>
+void inclusive_scan(Iter begin, Iter end, IterOut out)
+{
+  inclusive_scan(ExecPolicy{}, begin, end, out, std::plus<T>{}, T{0});
```

Comment:
I'm not sure if I'm missing something, but should `std::plus` be used here? Would that break if you used a CUDA execution policy?